straxen/plugins/bayes_peak_classification.py

```diff
@@ -121,30 +130,83 @@ def compute_inference(bins: int, bayes_n_nodes: int, cpt: np.ndarray, n_bayes_cl
     waveform_num_bin_edges = len(waveform_bin_edges)
     quantile_bin_edges = bins[1, :][bins[1, :] > -1]
     quantile_num_bin_edges = len(quantile_bin_edges)
-
     waveform_values = np.digitize(waveforms, bins=waveform_bin_edges)-1
-    waveform_values[waveform_values < 0] = int(0)
-    waveform_values[waveform_values > int(waveform_num_bin_edges - 2)] = int(waveform_num_bin_edges - 2)
+    waveform_values = np.clip(waveform_values, 0, waveform_num_bin_edges - 2)
 
     quantile_values = np.digitize(quantiles, bins=quantile_bin_edges)-1
-    quantile_values[quantile_values < 0] = int(0)
-    quantile_values[quantile_values > int(quantile_num_bin_edges - 2)] = int(quantile_num_bin_edges - 2)
+    quantile_values = np.clip(quantile_values, 0, quantile_num_bin_edges - 2)
+
+    lnposterior = get_log_posterior(bayes_n_nodes,
+                                    waveforms,
+                                    cpt,
+                                    waveform_num_bin_edges,
+                                    quantile_num_bin_edges,
+                                    n_bayes_classes,
+                                    waveform_values,
+                                    quantile_values,
+                                    )
 
-    values_for_inference = np.append(waveform_values, quantile_values, axis=1)
+    lnposterior_sumsamples = np.sum(lnposterior, axis=1)
+    lnposterior_sumsamples = lnposterior_sumsamples + np.log(class_prior)
+
+    # If you need to numbafy, use
+    # github.com/XENONnT/straxen/blob/c06a611a5a4289c65b9830fc1e68027149b6a5fd/straxen/plugins/bayes_peak_classification.py#L119  # noqa
+    normalization = logsumexp(lnposterior_sumsamples, axis=1)
+    ln_prob_s1 = lnposterior_sumsamples[:, 0]-normalization
+    ln_prob_s2 = lnposterior_sumsamples[:, 1]-normalization
+    return ln_prob_s1, ln_prob_s2
+
+
+def get_log_posterior(bayes_n_nodes: int,
+                      waveforms: np.ndarray,
+                      cpt: np.ndarray,
+                      waveform_num_bin_edges: int,
+                      quantile_num_bin_edges: int,
+                      n_bayes_classes: int,
+                      waveform_values: np.ndarray,
+                      quantile_values: np.ndarray,
+                      ) -> np.ndarray:
+    """
+    # TODO, add a description what we are computing here
```

Comment:
@ahiguera-mx can we get some information about how the log-posterior is computed? Maybe adding a few formulas would make things already more readable